ouroboros-consensus/test-storage/Test/Ouroboros/Storage/ChainDB/StateMachine.hs

```diff
@@ -1050,18 +1060,24 @@ prop_sequential = forAllCommands smUnused Nothing $ \cmds -> QC.monadicIO $ do
         realChain' <- toChain db'
         closeDB db'
 
-        ThreadRegistry.cancelAll registry
+        ThreadRegistry.cancelAll threadRegistry
+
+        -- closeDB should have closed all open Readers and Iterators, freeing
+        -- up all resources, so there should be no more clean-up actions left.
+        remainingCleanups <- ResourceRegistry.nbCleanups resourceRegistry
```

Comment:
Let's document that `remainingCleanups` is expected to be zero only because we don't currently test with any exceptions. If this was _always_ true the registry would not be necessary in the first place. 

Perhaps we _should_ actually modify the tests so that some exceptions are thrown, thereby testing the registry..? :thinking: